spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -363,6 +363,97 @@ def not_ok?; false; end
         expect(subject).to eq(3)
       end
     end
+
+    describe 'threadsafety', :threadsafe => true do
+      # Force call to eq to happen in threaded env to prevent "circular require" warnings
+      # Not actually sure why calling eq in two threads at once causes this (it's a sporadic failure):
```

Comment:
http://blog.plataformatec.com.br/2012/08/eager-loading-for-greater-good/ talks a bit about autoload being threadsafe on 2.0+.   Given it's supposedly threadsafe on 2.2, I'm surprised you're seeing issues, though...